src/types.rs

```diff
@@ -197,8 +205,20 @@ impl TryFrom<&Statement> for Command {
                                 entry.primary_key = *is_primary;
                                 entry.unique = true;
                             }
-                            ColumnOption::ForeignKey { .. } => {
-                                anyhow::bail!("FOREIGN KEY not yet supported")
+                            ColumnOption::ForeignKey {
+                                foreign_table,
+                                referred_columns,
+                                ..
+                            } => {
+                                if referred_columns.len() != 1 {
+                                    anyhow::bail!(
+                                        "Exactly one column must be specified for a foreign key"
+                                    );
+                                }
+                                entry.foreign_key = Some((
+                                    foreign_table.to_string(),
+                                    referred_columns[0].to_string(),
```

Comment:
For these index accessors would it be better to do something like:
```
referred_columns.get(0).unwrap_or_else(....);
```

to make it a bit more safe then doing a direct array access?

Reply:
There's an error check up above, so just gonna skip extra bounds checking